lib/statesman/machine.rb

```diff
@@ -163,6 +165,16 @@ def validate_from_and_to_state(from, to)
 
       private
 
+      def define_state_constant(state_name)
+        constant_name = state_name.upcase
+
+        if const_defined?(constant_name)
+          warn "Name conflict: '#{self.class.name}::#{constant_name}' is already defined"
```

Comment:
I think the potential for this happening is enough to raise an exception, that said it should be much less likely to occur if we include a module for the dynamic constants

Reply:
I agree.